### PR TITLE
Feat: login and FCM integration

### PR DESCRIPTION
**Pull request by gagansuie**

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -21,20 +21,29 @@ plugins {
...
-        targetSdk = 33
+        targetSdk = 34
         versionCode = 1
         versionName = "1.0"
-
-        testInstrumentationRunner = "io.codecrow.mage.HiltTestRunner"
+        addManifestPlaceholders(
+            hashMapOf(
+                ("auth0Domain" to "@string/com_auth0_domain"),
+                ("auth0Scheme" to "@string/com_auth0_scheme")
```
Comment (gagan-suie) [inline app/build.gradle.kts:41]: remove this.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -15,5 +15,12 @@
   -->
 
 <resources>
-    <string name="app_name">MageApplication</string>
+    <string name="app_name">Mage</string>
+    <string name="default_notification_channel_id">mage_notification_channel</string>
+
+    <string name="com_auth0_client_id">L3EhzWvwAfJyY1jNpGsSietpQ1ykvuWG</string>
+    <string name="com_auth0_domain">dev-mage-stream.us.auth0.com</string>
+    <string name="com_auth0_scheme">mage-mobile</string>
```
Comment (gagan-suie) [inline app/src/main/res/values/strings.xml:23]: Remove these

File: app/src/main/java/io/codecrow/mage/MageApplication.kt [hunk 3]
```diff
@@ -17,13 +17,24 @@
...
+        lateinit var auth0: Auth0
+            private set
+    }*/
+
     override fun onCreate() {
         super.onCreate()
+        /*auth0 = Auth0(
+            getString(R.string.com_auth0_client_id),
+            getString(R.string.com_auth0_domain)
+        )*/
```
Comment (gagan-suie) [inline app/src/main/java/io/codecrow/mage/MageApplication.kt:37]: Remove the commented code

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -15,5 +15,12 @@
...
+
+    <string name="appAuthRedirectScheme">https://discord.com/login?redirect_to=/oauth2/authorize?client_id=1025544065510555678&amp;redirect_uri=https%3A%2F%2Fapi.mage.stream%2Fauth%2Fdiscord%2Fcallback&amp;response_type=code&amp;scope=identify%20email&amp;state=eyJhcGlLZXkiOm51bGx9</string>
```
Comment (gagan-suie) [inline app/src/main/res/values/strings.xml:25]: Remove this

File: app/build.gradle.kts (hunk 1 above)
Comment (janak-jivrani) [reply app/build.gradle.kts]: removed this

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (janak-jivrani) [reply app/src/main/res/values/strings.xml]: removed this

File: app/src/main/java/io/codecrow/mage/MageApplication.kt (hunk 3 above)
Comment (janak-jivrani) [reply app/src/main/java/io/codecrow/mage/MageApplication.kt]: removed this

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (janak-jivrani) [reply app/src/main/res/values/strings.xml]: removed this

File: app/src/main/java/io/codecrow/mage/utils/Constant.kt [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package io.codecrow.mage.utils
+
+object Constant {
+    const val DiscordUrl = "https://discord.com/api/oauth2/authorize?client_id=1025544024129536030&redirect_uri=https%3A%2F%2Fdev.api.mage.stream%2Fauth%2Fdiscord%2Fcallback&response_type=code&scope=identify%20email&state=eyJhcGlLZXkiOiJjZG1hMnR3N2llem5rM25xanBjN2w3cmtnY3hyem0ifQ%3D%3D"
+    const val GoogleUrl = "https://accounts.google.com/o/oauth2/v2/auth?client_id=518394378279-9ifcegnc2jdi31g6mf7hrvvgcjcn6vd3.apps.googleusercontent.com&include_granted_scopes=true&redirect_uri=https%3A%2F%2Fdev.api.mage.stream%2Fauth%2Fgoogle%2Fcallback&response_type=code&scope=openid%20email%20profile&state=eyJhcGlLZXkiOiJjZG1hMnR3N2llem5rM25xanBjN2w3cmtnY3hyem0ifQ%3D%3D"
+    const val GithubUrl = "https://github.com/login/oauth/authorize?allow_signup=true&client_id=9ed2c795950e25904971&scope=read%3Auser%20user%3Aemail"
+
```
Comment (gagan-suie) [inline app/src/main/java/io/codecrow/mage/utils/Constant.kt:7]: Remove these hard-coded strings.

**Pull request closed without merging**